src/utils.cc

```diff
@@ -360,9 +364,20 @@ minio::utils::Time minio::utils::Time::FromISO8601UTC(const char* value) {
   return Time(time, tv_usec, true);
 }
 
+int minio::utils::Time::Compare(const Time& rhs) const {
```

Comment:
In all the relational operators that were broken. The original code compiled just because there was the implicit bool conversion operator. So when you were comparing e.g. `Time < Time`, the compiler failed to find the `<` operator, but then succeeded in casting both operands into `bool `and then returned the result of `bool < bool`, which is obviously wrong. Ditto with the `<<` operators.